collect_app/src/main/java/org/odk/collect/android/geo/GoogleMapFragment.java

```diff
@@ -846,4 +892,58 @@ private void clearPolyline() {
             }
         }
     }
+
+    private static class PolygonFeature implements MapFeature {
+
+        private Polygon polygon;
+        private final List<Marker> markers = new ArrayList<>();
+
+        PolygonFeature(Context context, GoogleMap map, Iterable<MapPoint> points, int strokeLineColor) {
+
+            for (MapPoint point : points) {
+                markers.add(createMarker(context, new MarkerDescription(point, false, CENTER, new MarkerIconDescription(R.drawable.ic_map_point)), map));
+            }
+
+            polygon = map.addPolygon(new PolygonOptions()
+                    .addAll(markers.stream().map(Marker::getPosition).collect(Collectors.toList()))
+                    .strokeColor(strokeLineColor)
+                    .strokeWidth(5)
+                    .fillColor(ColorUtils.setAlphaComponent(strokeLineColor, 150))
+                    .clickable(true)
+            );
+        }
+
+        @Override
+        public boolean ownsMarker(Marker marker) {
+            return markers.contains(marker);
```

Comment:
I think this could return false. As I understand this is for letting users click on one of the markers to select a polygon right? Is this needed they wanted to be able to click on the area inside and they have that plus they can still click on the lines. To me, it would be enough we don't need to provide clickability for all markers too.

Reply:
I found that if the markers weren't clickable, it was very hard to tap on a shape at the default zoom (unless it covered a country sized area).